### PR TITLE
308 remove guava precon

### PR DESCRIPTION
**Pull request by ricellis**
*What*

Replaced usages of Guava's `Preconditions` and `Strings`.

*How*

* Added `Misc` methods:
    * Added checkNotNull
    * Added checkNotNullOrEmpty
    * Added checkState
    * Added isStringNullOrEmpty
    * Added checkArgument
* Replaced `Precondition.checkNotNull` with `Misc.checkNotNull`:
    * Updated messages to only take a prefix.
    * Replaced Precondition with a  null check in the sample app instead of trying to use Guava or the internal utility.
* Replaced `Preconditions.checkArgument(!Strings.isNullOrEmpty(arg))` with
`Misc.checkNotNullOrEmpty`:
    * Updated messages to only take a prefix.
* Replaced usages of `Strings.isNullOrEmpty` with `Misc.isStringNullOrEmpty`
* Replaced `Precondtion.checkState` with `Misc.checkState`
* Replaced `Precondition.checkArgument` with `Misc.checkArgument`

*Testing*

No new tests, existing tests pass.

*Issues*

Part of #308

### CODE REVIEW

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/mazha/CouchClient.java [hunk 1]
```diff
@@ -406,7 +405,7 @@ public void putAttachmentStream(String id, String rev, String attachmentName, St
                                                       Collection<String> attsSince,
                                                       boolean pullAttachmentsInline) {
         Misc.checkNotNullOrEmpty(id, "id");
-        Preconditions.checkArgument(revisions.size() > 0, "Need at lease one open revision");
+        Misc.checkArgument(revisions.size() > 0, "Need at lease one open revision");
```
Comment (rhyshort) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/mazha/CouchClient.java]: While we are here we should correct the error message.

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ChangesResultWrapper.java [hunk 2]
```diff
@@ -50,9 +49,9 @@ public Object getLastSeq() {
     }
 
     public Multimap<String, String> openRevisions(int start, int end) {
-        Preconditions.checkArgument(start >= 0, "Start position must be greater or equal to zero.");
-        Preconditions.checkArgument(end > start, "End position must be greater than start.");
-        Preconditions.checkArgument(end <= this.size(), "End position must be smaller than changes feed size.");
+        Misc.checkArgument(start >= 0, "Start position must be greater or equal to zero.");
+        Misc.checkArgument(end > start, "End position must be greater than start.");
+        Misc.checkArgument(end <= this.size(), "End position must be smaller than changes feed size.");
```
Comment (rhyshort) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ChangesResultWrapper.java]: We need to correct this message too, `end position` can be equal to changes feed size.

**Comment by rhyshort** [comment]
Once those two error messages are correct +1

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/mazha/CouchClient.java (hunk 1 above)
Comment (ricellis) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/mazha/CouchClient.java]: Many of the changes were made by regex, so I didn't see the error messages - nice spot! Fixed in 122e842.

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ChangesResultWrapper.java (hunk 2 above)
Comment (ricellis) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ChangesResultWrapper.java]: Fixed in 122e842

**Pull request merged**